torch/csrc/autograd/variable.cpp

```diff
@@ -535,6 +536,8 @@ void handle_view_on_rebase(DifferentiableViewMeta* diff_view_meta, bool indirect
                        "version of the function that produced this view or "
                        "don't modify this view inplace.");
       } else {
+        TORCH_INTERNAL_ASSERT(creation_meta != CreationMeta::NO_VARIABLE_TYPE_VIEW,
```

Comment:
Why is this one internal?

Reply:
Oh it's because we error out for `NO_VARIABLE_TYPE_VIEW` in the beginning of VariableType kernel. Also tensors with `NO_VARIABLE_TYPE_VIEW` are view of normal tensors which means they have Autograd keys and will definitely go through VariableType kernel first. Thus we shouldn't expect to see NO_VARIABEL_TYPE_VIEW in as_view at all.

Reply:
This function can also be called from `.grad_fn` if another view was modified inplace and so a Tensor lazily rebase his history because someone is trying to access its grad_fn

Reply:
Hmmm right, but access to grad_fn should also have been blocked by the the check in the beginning of VariableType I assume? ;)